### PR TITLE
feat: add support for oauth storage providers

### PR DESCRIPTION
**Pull request by Panaetius**
This PR adds three new OAuth2 providers, Google Drive, Microsoft OneDrive and Dropbox.
I tested these with some hacked together code in this PR and after confirming they worked, undid the hacky changes. As such, this PR doesn't do anything. Or rather, it'd allow admins to set up OAuth2 apps and for users to connect to them, but once connected, the users can't use them anywhere.
There will be a follow-up build including UI changes that uses these changes to automatically generate tokens for cloud storage for users.

### CODE REVIEW

**Review by m-alisafaee — DISMISSED** [review]
LGTM, Thanks!

File: components/renku_data_services/errors/errors.py [hunk 1]
```diff
@@ -45,6 +45,16 @@ class UnauthorizedError(BaseError):
     quiet: bool = True
 
 
+@dataclass
+class InvalidOAuth2Token(BaseError):
```
Comment (leafty) [inline components/renku_data_services/errors/errors.py:49]: Why not have `class InvalidOAuth2Token(UnauthorizedError):` and only override the `message` field?

File: components/renku_data_services/migrations/versions/b8cbd62e85b9_add_google_drive_provider_kind.py
Comment (leafty) [inline components/renku_data_services/migrations/versions/b8cbd62e85b9_add_google_drive_provider_kind.py:1]: This migration should be collapsed with the next one.

**Review by leafty — APPROVED** [review]

File: components/renku_data_services/errors/errors.py [hunk 2]
```diff
@@ -45,6 +45,13 @@ class UnauthorizedError(BaseError):
     quiet: bool = True
 
 
+@dataclass
+class InvalidOAuth2Token(UnauthorizedError):
+    """Raised when a token for a connected service is invalid."""
+
+    message: str = "OAuth2 token for connected service invalid or expired."
```
Comment (leafty) [inline components/renku_data_services/errors/errors.py:52]: Non-blocking: does it make sense to add a subclass here when this seems to be used only once?

**Review by leafty — APPROVED** [review]
LGTM

**Review by leafty — DISMISSED** [review]

**Review by leafty — DISMISSED** [review]

**Comment by Panaetius** [comment]
I'd missed the new migration in the recent merge and had to consolidate migration heads

**Review by leafty — APPROVED** [review]

**Pull request merged**